server/src/ClientToDBAPI/UsersRoute/mainRoute.ts

```diff
@@ -141,26 +141,39 @@ usersRoute.get('/user', (request: Request, response: Response) => {
 });
 
 usersRoute.post('/changeInvestigator', adminMiddleWare, (request: Request, response: Response) => {
-    const changeInvestigatorVariables = {
-        epidemiologyNumber: request.body.epidemiologyNumber,
-        newUser: request.body.user
-    };
+    const epidemiologyNumbers: number[] = request.body.epidemiologyNumbers;
+    const newUser: string = request.body.user;
+    const transferReason: number = request.body.transferReason;
+
     logger.info({
         service: Service.SERVER,
         severity: Severity.LOW,
         workflow: 'Switch investigator',
-        step: `querying the graphql API with parameters ${JSON.stringify(changeInvestigatorVariables)}`,
+        step: `querying the graphql API with parameters ${JSON.stringify(request.body)}`,
         user: response.locals.user.id
     });
-    graphqlRequest(UPDATE_INVESTIGATOR, response.locals, changeInvestigatorVariables).then((result: any) => {
-        logger.info({
-            service: Service.SERVER,
-            severity: Severity.LOW,
-            workflow: 'Switch investigator',
-            step: `investigator have been changed in the DB, investigation: ${request.body.epidemiologyNumber}`,
-            user: response.locals.user.id
-        });
-        response.send(result.data)
+    Promise.all(epidemiologyNumbers.map(epidemiologyNumber => graphqlRequest(UPDATE_INVESTIGATOR, response.locals, 
+        {epidemiologyNumber, newUser, transferReason})))
+    .then((results: any[]) => {
+        if (areAllResultsValid(results)) {
+            logger.info({
+                service: Service.SERVER,
+                severity: Severity.LOW,
+                workflow: 'Switch investigator',
+                step: `investigator have been changed in the DB, investigations: ${epidemiologyNumbers}`,
+                user: response.locals.user.id
+            });
+            response.send(results[0]?.data || '');
```

Comment:
in what case the data would be empty null? Maybe you should make the check for errors also according to this condition